### PR TITLE
skip person selection if immunization person already exists #7160

### LINKED ISSUE #7160: Person select inconsistent when creating an immunization
### Bug Description

When creating a new immunization while editing a case, the person select behaviour is inconsistent. The case person is offered for selection, but selecting this person and clicking "select" results in an error.
However, not selecting a person and clicking "create" works.

### Steps to Reproduce

1. Navigate to a case, create a new immunization, save
2. The person select window pops up. Select the case person, click "select"

### Expected Behavior

When creating an immunization for an existing case, the person select window should not appear in the first pace, since the person in question is defined (the case's person).
If there are technical or other reasons not to skip the person select window, the case person should be selectable without errors.

### Screenshots

image

### System Details

* Device: Android
* SORMAS version: 1.64.1
* Android version/Browser: API 24
* Server URL: https://release-international.sormas.netzlink.com
* User Role:

### PR DESCRIPTION
**Pull request by syntakker**
Fixes #7160

### CODE REVIEW

File: sormas-app/app/src/main/java/de/symeda/sormas/app/person/SelectOrCreatePersonDialog.java [hunk 1]
```diff
@@ -72,7 +72,7 @@ public static void selectOrCreatePerson(final Person person, final Consumer<Pers
 		}
 
 		personDialog.setPositiveCallback(() -> {
-			if (personDialog.getSelectedPerson() != null && !personDialog.getSelectedPerson().getUuid().equals(person.getUuid())) {
+			if (personDialog.getSelectedPerson() != null) {
```
Comment (syntakker) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/person/SelectOrCreatePersonDialog.java:75]: `personDialog.getSelectedPerson().getUuid().equals(person.getUuid())` should never be true. If the person already existed before, person selection should be skipped.
If - under any circumstances - a person with the same UUID is selected from a list of already existing ones, a merge should take place here.

File: sormas-app/app/src/main/java/de/symeda/sormas/app/immunization/edit/ImmunizationNewFragment.java [hunk 2]
```diff
@@ -252,5 +252,16 @@ protected void onAfterLayoutBinding(FragmentImmunizationNewLayoutBinding content
...
+		Bundler bundler = new Bundler(getArguments());
+		if (bundler.getCaseUuid() != null || bundler.getContactUuid() != null || bundler.getEventParticipantUuid() != null) {
+			contentBinding.personFirstName.setEnabled(false);
+			contentBinding.personLastName.setEnabled(false);
+			contentBinding.personSex.setEnabled(false);
+			contentBinding.personBirthdateYYYY.setEnabled(false);
+			contentBinding.personBirthdateMM.setEnabled(false);
+			contentBinding.personBirthdateDD.setEnabled(false);
+		}
 	}
```
Comment (syntakker) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/immunization/edit/ImmunizationNewFragment.java:266]: As for cases, see `CaseNewFragment`, l.304ff:
```java
		// Disable personal details and disease fields when case is created from contact
		// or event person
		Bundler bundler = new Bundler(getArguments());
		if (bundler.getContactUuid() != null || bundler.getEventParticipantUuid() != null) {
...
```

File: sormas-app/app/src/main/java/de/symeda/sormas/app/immunization/edit/ImmunizationNewActivity.java [hunk 3]
```diff
@@ -178,12 +178,17 @@ public void saveData() {
...
+			pickOrCreateImmunizationAndSave(immunization, fragment);
+		} else {
+			SelectOrCreatePersonDialog.selectOrCreatePerson(immunization.getPerson(), person -> {
+				if (person != null) {
+					immunization.setPerson(person);
+					pickOrCreateImmunizationAndSave(immunization, fragment);
+				}
+			});
+		}
 	}
```
Comment (syntakker) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/immunization/edit/ImmunizationNewActivity.java:192]: As for cases, see `CaseNewActivity`, l.208:
```java
		// Person selection can be skipped if the case was created from a contact
		if (contactUuid == null) {
...
```

**Review by MateStrysewske — APPROVED** [review]
Code is approved, but I can't test it because I'm already on Arctic Fox (which is incompatible with the gradle version used on the hotfix branch). Please assign another developer to verify that everything works correctly.

**Review by alexcaruntu-vita — APPROVED** [review]

**Pull request merged**